### DISCUSSION TITLE
[Question] Cannot access services that run on the Host of wg-easy

### DISCUSSION THREAD

**Discussion by iHaveAstream** [wg-easy/wg-easy#1358] [Q&A]
Hi,

I'm running ```wg-easy``` on my Synology NAS (DSM 7.2) in a docker environment.
Everything is fine, which means that I can access internet thru the VPN and can also access devices of different VLANs at home.
Except one thing does not work: Access to services that run on the Host of ```wg-easy``` - the Synology NAS.

This is my wg0.conf
```
# Note: Do not edit this file directly.
# Your changes will be overwritten!

# Server
[Interface]
PrivateKey = ************************
Address = 10.8.0.1/24
ListenPort = 51820
PreUp = 
PostUp =  iptables -t nat -A POSTROUTING -s 10.8.0.0/24 -o eth0 -j MASQUERADE; iptables -A INPUT -p udp -m udp --dport 51820 -j ACCEPT; iptables -A FORWARD -i wg0 -j ACCEPT; iptables -A FORWARD -o wg0 -j ACCEPT; 
PreDown = 
PostDown =  iptables -t nat -D POSTROUTING -s 10.8.0.0/24 -o eth0 -j MASQUERADE; iptables -D INPUT -p udp -m udp --dport 51820 -j ACCEPT; iptables -D FORWARD -i wg0 -j ACCEPT; iptables -D FORWARD -o wg0 -j ACCEPT; 


# Client: A
[Peer]
PublicKey = ****************************
PresharedKey = *************************
AllowedIPs = 10.8.0.5/32
```

And this is the ClientA.conf
```
[Interface]
PrivateKey = *******************************
Address = 10.8.0.5/24
DNS = 10.10.10.2

[Peer]
PublicKey = ********************************
PresharedKey = *********************************
AllowedIPs = 0.0.0.0/0
PersistentKeepalive = 25
Endpoint = my.wg.vpn
```

The Synology NAS which is the Host of the ```wg-easy``` Docker instance has one physical NIC (eth0).
But I'm running two virtual adapters: eth0.20 (VLAN 192.168.20.0/24 with IP 192.168.20.250) and eth0.40 (VLAN 10.10.40.0/24 with IP 10.10.40.250).
When my Client is now connected thru ```wg-easy```, I cannot access any service that runs on the Host - even not the DSM Web UI under https://192.168.20.250:5001 or https://10.10.40.250:5001
But, I can access other services available in those two VLANs which are running on other VMs.

What do I miss in my ```wg-easy``` config?

Thanks!

**Comment by shalafi99** [comment]
Hi, considering you can access any other IP addresses/services you have internally except for the DSM's own IPs, it seems unexpected that the wireguard config would be the culprit.

Nevertheless, I checked my config and noticed two differences, not sure if it can help you but here we go:

1) any traffic is being NAT'ed "iptables -t nat -A POSTROUTING -o eth0 -j MASQUERADE;", instead of only what originates from the wireguard subnet
2) also includes "iptables -A FORWARD -m state --state ESTABLISHED,RELATED -j ACCEPT;" for state tracking

I am not familiar with network configuration options available for a Synology NAS but I am assuming it is not blocking incoming connections originating from 10.8.0.0/24.

Cheers

**Comment by iHaveAstream** [comment]
HI @shalafi99 
thanks for your reply. I think it might be worth a try.
Could you possibly paste your complete ```WG_POST_UP``` and ```WG_POST_DOWN``` config option pls.?

**Reply by shalafi99** [reply]
Hi, sure can.
But please keep in mind my own config is quite restrictive and this can make it somewhat harder to read...
(_Access is allowed only to this 192.168.2.160 internal host and essentially only for DNS traffic + reverse proxying of https/tcp/udp)_.

`WG_POST_UP=iptables -A FORWARD -m state --state ESTABLISHED,RELATED -j ACCEPT; iptables -A FORWARD -i wg0 -s 192.168.5.0/24 -d 192.168.2.160/32 -p tcp --dport 443 -j ACCEPT; iptables -A FORWARD -i wg0 -s 192.168.5.0/24 -d 192.168.2.160/32 -p udp --dport 53 -j ACCEPT; iptables -A FORWARD -i wg0 -s 192.168.5.0/24 -d 192.168.2.160/32 -p tcp --dport 53 -j ACCEPT; iptables -A FORWARD -i wg0 -s 192.168.5.0/24 -d 192.168.2.160/32 -p tcp --dport 853 -j ACCEPT; iptables -A FORWARD -i wg0 -s 192.168.5.0/24 -d 192.168.2.160/32 -p tcp --dport 3389 -j ACCEPT; iptables -A FORWARD -i wg0 -s 192.168.5.0/24 -d 192.168.2.160/32 -p udp --dport 9 -j ACCEPT; iptables -A FORWARD -i wg0 -s 192.168.5.0/24 -d 192.168.2.160/32 -p icmp -j ACCEPT; iptables -A FORWARD -i wg0 -j DROP; iptables -t nat -A POSTROUTING -o eth0 -j MASQUERADE`

`WG_POST_DOWN=iptables -D FORWARD -m state --state ESTABLISHED,RELATED -j ACCEPT; iptables -D FORWARD -i wg0 -s 192.168.5.0/24 -d 192.168.2.160/32 -p tcp --dport 443 -j ACCEPT; iptables -D FORWARD -i wg0 -s 192.168.5.0/24 -d 192.168.2.160/32 -p udp --dport 53 -j ACCEPT; iptables -D FORWARD -i wg0 -s 192.168.5.0/24 -d 192.168.2.160/32 -p tcp --dport 53 -j ACCEPT; iptables -D FORWARD -i wg0 -s 192.168.5.0/24 -d 192.168.2.160/32 -p tcp --dport 853 -j ACCEPT; iptables -D FORWARD -i wg0 -s 192.168.5.0/24 -d 192.168.2.160/32 -p tcp --dport 3389 -j ACCEPT; iptables -D FORWARD -i wg0 -s 192.168.5.0/24 -d 192.168.2.160/32 -p udp --dport 9 -j ACCEPT; iptables -D FORWARD -i wg0 -s 192.168.5.0/24 -d 192.168.2.160/32 -p icmp -j ACCEPT; iptables -D FORWARD -i wg0 -j DROP; iptables -t nat -D POSTROUTING -o eth0 -j MASQUERADE`

*edit* complementing: my edge port forwarding rule used for wireguard is mapped internally to this same 192.168.2.160 IP (wg-easy and the reverse proxy are separate containers with their port mappings to this IP, DNS is running directly on the host itself)

Something else that occurred to me, as an attempt to try to isolate the issue between wg-easy and the NAS device itself (_but it would depend on what resources you have at your disposal with the current environment_), was to spin up this same config in another docker host and temporarily switch your edge port forwarding to it. And by doing so, the whole point would be to then attempt to access some service running on that other docker host (instead of the Synology). Could be a lot of work for one test.

**Comment by serambca** [comment]
Hi, 
I need some help.
If I use a docker container, I can connect to server, access to admin website and Internet across the server but I can't see nothing in ifconfig due I understand that works inside the lan of docker.

My problem is that I need to access to the services calling by IP of wireguard. For example, access to webserver that I have in the server by internal IP address but I can't access. 

Could you please help me to understand well how can I solve?

Best regards!!
